trace/doc.go

```diff
@@ -0,0 +1,134 @@
+// Package trace provies an API for initiating and reporting SSF
```

Comment:
This is good and clarifies a few things.

A few areas you may want to elaborate on:

* People are confused by Context sometimes and the fact that it carries the spans you just created, and should be passed through your application. Maybe a blurb about this or link to outside tutorial?

* Another thing I'm missing is how to extract a span from a context without starting a new one.

This is useful if a.) I don't want to dilute my trace with in-process spans for every layer b.) I have a helper function that does something with traces.

A good example is traceLogger from my recent PR:

```
// traceLogger returns a logrus Entry with trace_id and span_id affixed so the log entries can be unified with traces.
func traceLogger(log *logrus.Logger, ctx context.Context) *logrus.Entry {
	if span, ok := opentracing.SpanFromContext(ctx).(*trace.Span); ok {
		return log.
			WithField("trace_id", span.TraceID).
			WithField("span_id", span.SpanID)
	}
	return log.WithField("trace_id", "<UNKNOWN>")
}
```

Reply:
Good suggestions, I'll incorporate them!